### ISSUE TITLE
[fusion-cli#426] Modern bundles for modern browsers

### ISSUE THREAD

**Issue by chrisdothtml** [fusionjs/fusionjs#326]
This issue was migrated from fusion-cli and was originally reported by @ksheedlo.

---

### Type of issue

Feature request

### Description

It would be cool if, instead of compiling all ES6 class syntax, async/await etc to ES5, Fusion could detect modern browsers that support these features and send them a more optimized bundle. For instance, class syntax doesn't need to be compiled into functions and prototypes anymore. This could significantly reduce the size of bundles sent to browsers that support modern features, since code volume generally goes up to support the newer syntax in the legacy environment.

Rel #34 

### Current behavior

Fusion delivers code to all browsers using the same compiler settings.

### Desired behavior

Fusion delivers code to modern browsers using more aggressive compiler settings that allow browsers to provide their own class syntax and other ES6 features. Fusion still delivers the same transpiled and polyfilled code to legacy browsers.

**Comment by brendean** [comment]
Is there any progress on this issue? I was looking into bumping moder browser versions and the thing holding back reduction of core-js polyfills included as part of fusion-cli is iOS and Safari being listed as modern (this is via the `targets` babel config).

Removing iOS and Safari and bumping the version of Chrome/Android/Firefox reduces the vendor bundle by 159.13 KB (uncompressed).

Its not desirable to serve the legacy bundle to iOS/Safari so it seems there might need to be 3 classes of browser bundles (legacy, "mainstream", modern). Would like to hear feedback on this idea before opening a PR.

**Comment by rtsao** [comment]
So even Safari 13/14 requires too many polyfills?

I think one option is switching over to https://github.com/babel/preset-modules for the modern build (and adjusting the browser useragents accordingly), which does a better job of working around browser bugs. Then we should be able to include more browsers in the modern build and have fewer polyfills in the modern build.

**Comment by brendean** [comment]
Yep surprisingly (to me at least) including iOS/Safari 14 negated most of the drop that bumping the Chrome/Android/Firefox versions introduced: 42.31 KB instead of 159.13 KB (uncompressed).
